File: JAVA/Calendar.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Calendar {
    private Map<Integer, List<Integer>> dayMeetings;

    public Calendar() {
        dayMeetings = new HashMap<>();
    }

    public void addMeeting(int day, int meetingID) {
        // Check if the day exists in the calendar, if not, create it
        if (!dayMeetings.containsKey(day)) {
            dayMeetings.put(day, new ArrayList<>());
        }

        // Add the meeting ID to the specified day
        dayMeetings.get(day).add(meetingID);
    }

    public List<Integer> getMeetings(int day) {
        if (dayMeetings.containsKey(day)) {
            return dayMeetings.get(day);
        }
        return new ArrayList<>(); // Return an empty list if no meetings for the day
    }

    public void display(int day) {
        for (int meetingId : dayMeetings.get(day)) {
            MeetingManager meetingMgr = MeetingManager.getInstance();
            Meeting meeting = meetingMgr.getMeeting(meetingId);
            meeting.display();
        }
    }
}

File: C++/calendar.cpp
#include "Calendar.hpp"

Calendar::Calendar() {}

void Calendar::addMeeting(int day, int meetingID) {
    // Check if the day exists in the calendar, if not, create it
    if (dayMeetings.find(day) == dayMeetings.end()) {
        dayMeetings[day] = vector<int>();
    }

    // Add the meeting ID to the specified day
    dayMeetings[day].push_back(meetingID);
}

vector<int> Calendar::getMeetings(int day) const {
    if (dayMeetings.find(day) != dayMeetings.end()) {
        return dayMeetings.at(day);
    }
    return vector<int>(); // Return an empty vector if no meetings for the day
}

void Calendar::display(int day) {
    for(auto meetingId : dayMeetings[day]) {
        MeetingManager meetingMgr = MeetingManager::getInstance();
        Meeting* meeting = meetingMgr.getMeeting(meetingId);
        meeting->display();
    }
}
